app-editors/helix/helix-23.05.ebuild

```diff
@@ -0,0 +1,322 @@
+	redox_syscall-0.3.5
+	regex-1.8.1
+	regex-automata-0.1.10
+	regex-syntax-0.6.29
+	regex-syntax-0.7.1
+	ropey-1.6.0
+	rustix-0.37.15
+	ryu-1.0.13
+	same-file-1.0.6
+	scopeguard-1.1.0
+	scratch-1.0.5
+	serde-1.0.163
+	serde_derive-1.0.163
+	serde_json-1.0.96
+	serde_repr-0.1.12
+	serde_spanned-0.6.1
+	sha1_smol-1.0.0
+	signal-hook-0.3.15
+	signal-hook-mio-0.2.3
+	signal-hook-registry-1.4.1
+	signal-hook-tokio-0.3.1
+	slab-0.4.8
+	slotmap-1.0.6
+	smallvec-1.10.0
+	smartstring-1.0.1
+	smawk-0.3.1
+	socket2-0.4.9
+	static_assertions-1.1.0
+	str-buf-1.0.6
+	str_indices-0.4.1
+	syn-1.0.109
+	syn-2.0.15
+	tempfile-3.5.0
+	termcolor-1.2.0
+	termini-1.0.0
+	textwrap-0.16.0
+	thiserror-1.0.40
+	thiserror-impl-1.0.40
+	thread_local-1.1.7
+	threadpool-1.8.1
+	time-0.3.20
+	time-core-0.1.0
+	time-macros-0.2.8
+	tinyvec-1.6.0
+	tinyvec_macros-0.1.1
+	tokio-1.28.1
+	tokio-macros-2.1.0
+	tokio-stream-0.1.14
+	toml-0.7.3
+	toml_datetime-0.6.1
+	toml_edit-0.19.8
+	tree-sitter-0.20.10
+	unicase-2.6.0
+	unicode-bidi-0.3.13
+	unicode-bom-2.0.2
+	unicode-general-category-0.6.0
+	unicode-ident-1.0.8
+	unicode-linebreak-0.1.4
+	unicode-normalization-0.1.22
+	unicode-segmentation-1.10.1
+	unicode-width-0.1.10
+	url-2.3.1
+	version_check-0.9.4
+	walkdir-2.3.3
+	wasi-0.11.0+wasi-snapshot-preview1
+	wasm-bindgen-0.2.84
+	wasm-bindgen-backend-0.2.84
+	wasm-bindgen-macro-0.2.84
+	wasm-bindgen-macro-support-0.2.84
+	wasm-bindgen-shared-0.2.84
+	which-4.4.0
+	winapi-0.3.9
+	winapi-i686-pc-windows-gnu-0.4.0
+	winapi-util-0.1.5
+	winapi-x86_64-pc-windows-gnu-0.4.0
+	windows-0.48.0
+	windows-sys-0.45.0
+	windows-sys-0.48.0
+	windows-targets-0.42.2
+	windows-targets-0.48.0
+	windows_aarch64_gnullvm-0.42.2
+	windows_aarch64_gnullvm-0.48.0
+	windows_aarch64_msvc-0.42.2
+	windows_aarch64_msvc-0.48.0
+	windows_i686_gnu-0.42.2
+	windows_i686_gnu-0.48.0
+	windows_i686_msvc-0.42.2
+	windows_i686_msvc-0.48.0
+	windows_x86_64_gnu-0.42.2
+	windows_x86_64_gnu-0.48.0
+	windows_x86_64_gnullvm-0.42.2
+	windows_x86_64_gnullvm-0.48.0
+	windows_x86_64_msvc-0.42.2
+	windows_x86_64_msvc-0.48.0
+	winnow-0.4.1
+"
+
+inherit bash-completion-r1 cargo desktop xdg-utils
+
+DESCRIPTION="A post-modern text editor."
+HOMEPAGE="
+	https://helix-editor.com/
+	https://github.com/helix-editor/helix
+"
+SRC_URI="
+	https://github.com/helix-editor/helix/releases/download/${PV}/${P}-source.tar.xz -> ${P}.tar.xz
+	$(cargo_crate_uris)
+"
+
+S="${WORKDIR}"
+
+LICENSE="0BSD Apache-2.0 Apache-2.0-with-LLVM-exceptions BSD Boost-1.0 ISC MIT MPL-2.0 Unicode-DFS-2016 Unlicense ZLIB"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="doc +grammar"
+
+QA_FLAGS_IGNORED="
+	usr/bin/hx
+	usr/share/helix/runtime/grammars/.*\.so
+"
+
+DOCS=(
+	README.md
+	CHANGELOG.md
+	book/
+	docs/
+)
+
+src_compile() {
+	use grammar || local -x HELIX_DISABLE_AUTO_GRAMMAR_BUILD=1
+
+	cargo_src_compile
+}
+
+src_install() {
+	cargo_src_install --path helix-term
+
+	rm -rf runtime/grammars/.gitkeep || die
+	rm -rf runtime/grammars/sources || die
```

Comment:
Drop the `-f`s. They silently fail leaving possibly redundant calls in the ebuild for ages.